src/components/media-player/dialog-join-media-players.ts

```diff
@@ -0,0 +1,207 @@
+
+    this._selectedEntities = this._groupMembers;
+  }
+
+  public closeDialog() {
+    this._entityId = undefined;
+    this._selectedEntities = [];
+    this._submitting = false;
+    this._error = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._entityId) {
+      return nothing;
+    }
+
+    const entityId = this._entityId;
+    const stateObj = this.hass.states[entityId] as HassEntity | undefined;
+    const name = (stateObj && computeStateName(stateObj)) || entityId;
+
+    return html`
+      <ha-dialog
+        open
+        scrimClickAction
+        escapeKeyAction
+        flexContent
+        .heading=${name}
+        @closed=${this.closeDialog}
+      >
+        <ha-dialog-header show-border slot="heading">
+          <ha-icon-button
+            .label=${this.hass.localize("ui.common.close")}
+            .path=${mdiClose}
+            dialogAction="close"
+            slot="navigationIcon"
+          ></ha-icon-button>
+          <span slot="title"
+            >${this.hass.localize("ui.card.media_player.media_players")}</span
+          >
+          <ha-button slot="actionItems" @click=${this._selectAll}>
+            ${this.hass.localize("ui.card.media_player.select_all")}
+          </ha-button>
+        </ha-dialog-header>
+        ${this._error
+          ? html`<ha-alert alert-type="error">${this._error}</ha-alert>`
+          : nothing}
+        <div class="content">
+          <ha-media-player-toggle
+            .hass=${this.hass}
+            .entityId=${entityId}
+            checked
+            disabled
+          ></ha-media-player-toggle>
+          ${this._mediaPlayerEntities(this.hass.entities).map(
+            (entity) =>
+              html`<ha-media-player-toggle
+                .hass=${this.hass}
+                .entityId=${entity.entity_id}
+                .checked=${this._selectedEntities.includes(entity.entity_id)}
+                @change=${this._handleSelectedChange}
+              ></ha-media-player-toggle>`
+          )}
+        </div>
+        <ha-button slot="secondaryAction" @click=${this.closeDialog}>
+          ${this.hass.localize("ui.common.cancel")}
+        </ha-button>
+        <ha-button
+          .disabled=${this._submitting}
+          slot="primaryAction"
+          @click=${this._submit}
+        >
+          ${this.hass.localize("ui.common.apply")}
+        </ha-button>
+      </ha-dialog>
+    `;
+  }
+
+  private _mediaPlayerEntities = (
+    entities: Record<string, EntityRegistryDisplayEntry>
+  ) =>
+    Object.values(entities).filter(
+      (entity) =>
+        computeDomain(entity.entity_id) === "media_player" &&
+        this._entityId !== undefined &&
+        entity.entity_id !== this._entityId &&
+        this.hass.entities[entity.entity_id].platform ===
+          this.hass.entities[this._entityId].platform &&
+        supportsFeature(
+          this.hass.states[entity.entity_id],
```

Comment:
Not every entity in the entity registry has to be in the state machine, we should check that first